src/Collections/TerminusCollection.php

```diff
@@ -91,6 +94,22 @@ public function all()
     public function fetch()
     {
         foreach ($this->getData() as $id => $model_data) {
+            if (!is_object($model_data)) {
+                // For some reason I can't replicate, occasionally $model_data is just a string here.
+                $trace = debug_backtrace();
+                $error_message = "Fetch failed {file}:{line} model_data expected as object but returned as {type}.";
+                $context = [
+                    'file' => $trace[0]['file'],
+                    'line' => $trace[0]['line'],
+                    'type' => gettype($model_data),
+                ];
+                if (is_string($model_data)) {
+                    $error_message .= " String value: {string}";
+                    $context['string'] = strlen($model_data) > 250 ? substr($model_data, 0, 250) . '...' : $model_data;
+                }
+                $this->logger->error($error_message, $context);
+                break;
+            }
```

Comment:
I'm on board with this.  I might even change 106 to `is_scalar` and add an else that gathered the value via `$error_message .= " Value: " . print_r($model_data, true);` .. but there might be a chance of `print_r()` out of memory'ing...